ext/standard/mail.c

```diff
@@ -221,6 +221,41 @@ void php_mail_log_to_file(char *filename, char *message, size_t message_size TSR
 }
 
 
+static int php_mail_detect_multiple_crlf(char *hdr) {
+	/* This function detects malformed multiple newlines also */
+	char *tmp = hdr;
+
+	/* Should not have any newlines at the beginning */
+	while (*hdr && (*hdr == '\n' || *hdr == '\r')) {
+		hdr++;
+	}
+	if (tmp != hdr) {
+		return 1;
+	}
+
+	while(*hdr) {
+		if (*hdr == '\r') {
+			if (*(hdr+1) == '\r' || (*(hdr+1) == '\n' && (*(hdr+2) == '\n' || *(hdr+2) == '\r'))) {
+				/* Malformed or multiple newlines */
+				return 1;
+			} else {
+				hdr += 2;
```

Comment:
Right. I'm being sloppy here by assuming caller trimmed leading/ending \r, \n.
I may get rid of this assumption and previously applied trim code.